src/utils/utils.jl

```diff
@@ -37,27 +63,44 @@ function clear_all_serialized_system()
     return
 end
 
-function get_serialization_dir(has_forecasts::Bool, has_reserves::Bool)
-    if has_forecasts && has_reserves
-        return SERIALIZE_FORECASTRESERVE_DIR
-    elseif has_forecasts
-        return SERIALIZE_FORECASTONLY_DIR
-    elseif has_reserves
-        return SERIALIZE_RESERVEONLY_DIR
+function get_serialization_dir(case_args::Dict{Symbol, <:Any} = Dict{Symbol, Any}())
+    args_string = join(["$key=$value" for (key, value) in case_args], "_")
+    hash_value = hash(args_string)
+    return joinpath(PACKAGE_DIR, "data", "serialized_system", "$hash_value")
+end
+
+#make sure to have a check for unique name
+function get_serialized_filepath(
+    name::String,
+    case_args::Dict{Symbol, <:Any} = Dict{Symbol, Any}(),
+)
+    dir = get_serialization_dir(case_args)
+    if isdir(dir) && has_duplicates(dir, "$(name).json")
```

Comment:
Consider saving the filename into a variable so that you don’t have to repeat the string concatenation.